widget/embedded/src/utils/wallets.ts

```diff
@@ -380,36 +380,52 @@ export function formatBalance(balance: Balance | null): Balance | null {
   return formattedBalance;
 }
 
-export function sortTokensByBalance(
+function sortTokensByBalance(
+  token1Balance: Balance | null,
+  token2Balance: Balance | null
+): number {
+  if (token1Balance?.usdValue && token2Balance?.usdValue) {
+    return (
+      parseFloat(token2Balance.usdValue) - parseFloat(token1Balance.usdValue)
+    );
+  }
+
+  if (!token1Balance?.usdValue && token2Balance?.usdValue) {
+    return 1;
+  }
+
+  if (token1Balance?.usdValue && !token2Balance?.usdValue) {
+    return -1;
+  }
+
+  if (!token1Balance?.usdValue && !token2Balance?.usdValue) {
+    return (
+      parseFloat(token2Balance?.amount || '0') -
+      parseFloat(token1Balance?.amount || '0')
+    );
+  }
+
+  return 0;
+}
+
+export function sortTokens(
   tokens: Token[],
-  getBalanceFor: (token: Token) => Balance | null
+  getBalanceFor: (token: Token) => Balance | null,
+  isTokenPinned: (token: Token) => boolean
 ) {
   tokens.sort((token1, token2) => {
     const token1Balance = getBalanceFor(token1);
     const token2Balance = getBalanceFor(token2);
-
-    if (token1Balance?.usdValue && token2Balance?.usdValue) {
-      return (
-        parseFloat(token2Balance.usdValue) - parseFloat(token1Balance.usdValue)
-      );
-    }
-
-    if (!token1Balance?.usdValue && token2Balance?.usdValue) {
-      return 1;
-    }
-
-    if (token1Balance?.usdValue && !token2Balance?.usdValue) {
-      return -1;
-    }
-
-    if (!token1Balance?.usdValue && !token2Balance?.usdValue) {
-      return (
-        parseFloat(token2Balance?.amount || '0') -
-        parseFloat(token1Balance?.amount || '0')
-      );
+    const isToken1Pinned = isTokenPinned(token1);
+    const isToken2Pinned = isTokenPinned(token2);
+    // Check pinned tokens
+    const isPin =
+      isToken1Pinned === isToken2Pinned ? 0 : isToken1Pinned ? -1 : 1;
+    if (isPin !== 0) {
+      return isPin;
```

Comment:
I believe it would be more beneficial to relocate this logic to a separate helper function. Consolidating all sort helpers in this location and eliminating nested ternary operations would enhance readability.

Reply:
fixed